pkg/api/api_implementation.go

```diff
@@ -359,16 +359,25 @@ func (a defaultAPI) doRequest(req *http.Request, obj types.Object, body interfac
 		return err
 	}
 
-	if mediaType, err := getResponseType(response); err == nil {
-		if mediaType == "application/json" {
-			return json.NewDecoder(response.Body).Decode(body)
-		}
-
-		// unreachable, getResponseType() already checks for supported types
-		return ErrUnsupportedResponseFormat
-	} else {
+	emptyResponse, responseBodyReader, err := checkEmptyReader(response.Body)
+	if err != nil {
 		return err
 	}
+
+	if !emptyResponse {
```

Comment:
I think it is kind of the wrong way around if we first check the body, and afterwards the media type - shouldn't we first look at the media type, and then based on that decide what to do with the content?
There could be other mediaTypes where an empty body is completely unacceptable and we need to return an error.

Reply:
sadly the engine sends a content-type of application/json together with a `204 No Content` .. so that does not work